aten/src/ATen/native/mkldnn/MkldnnTensorMath.cpp

```diff
@@ -0,0 +1,49 @@
+#include <ATen/ATen.h>
+#include <ATen/Config.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/Parallel.h>
+#include <ATen/cpu/vec256/functional.h>
+#include <ATen/cpu/vec256/vec256.h>
+
+#if !AT_MKLDNN_ENABLED()
+
+namespace at {
+namespace native {
+
+Tensor& mkldnn_zero_(Tensor& self) {
+  AT_ERROR("mkldnn_zero_: ATen not compiled with MKLDNN support");
+}
+
+} // namespace native
+} // namespace at
+
+#else // AT_MKLDNN_EBABLED
+
+#include <ATen/native/mkldnn/MKLDNNCommon.h>
+
+namespace at {
+namespace native {
+
+Tensor& mkldnn_zero_(Tensor& self) {
+  using Vec = vec256::Vec256<float>;
+
+  ideep::tensor& x = itensor_from_mkldnn(self);
+
+  auto n = x.get_nelems();
+  auto* x_ = static_cast<float *>(x.get_data_handle());
+  parallel_for(0, n, 2048, [x_](int64_t begin, int64_t end){
+    vec256::map(
```

Comment:
@cpuhrsch Is it safe to unconditionally using these vectorized utilities? Or do we need to guard it by some avx macros check?

Reply:
Vec256 uses __AVX__ and __AVX2__ guards to compile with the compatibility version. So calling https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cpu/vec256/functional.h#L99 without compiling it for avx2 is correct.

But, since this isn't compiled with avx2 for OSS, it might be slow for users. Why isn't this a kernel?

Reply:
@cpuhrsch , if written as a kernel, the input should not be a TensorIterator  as [add_kernel](https://github.com/pytorch/pytorch/blob/cbf572671d4c46fa7bd5cd0f8a01ca40f623d6c6/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp#L16) , or written as [SoftMaxKernel](https://github.com/pytorch/pytorch/blob/cbf572671d4c46fa7bd5cd0f8a01ca40f623d6c6/aten/src/ATen/native/cpu/SoftMaxKernel.cpp#L25), please give me some advice, thanks!

Reply:
cc @VitalyFedyunin

Reply:
Inputs doesn't have to be TensorIterator. We prefer TensorIterator as input for kernels as we use same iterator for CPU and CUDA. In your case you can implement kernel that takes Tensor as input.

Reply:
Using map here is also inefficient, please consider writing something like vec256::fill which will loadu vector, fill it and store.

Reply:
@VitalyFedyunin , thanks for your suggestion, I will change the code.

Reply:
@VitalyFedyunin , I have added a fill API in ideep, perharps direct process opaque tensor is not good in pytorch,  After talking with @jgong5,  add soomthing like vec256::fill is only useful mkldnn tensor, perhaps this way have not big benefit. Thanks!